test/shape-modes.html

```diff
@@ -0,0 +1,183 @@
+    const MODES = [ 'corners', 'corner', 'center', 'radius' ];
+    let SHAPE = 'ellipse';
+    let MODE = 'corners';
+    let num = 1;
+
+    function nextSetting() {
+        let modeIdx = MODES.indexOf(MODE);
+        let shapeIdx = SHAPES.indexOf(SHAPE);
+        modeIdx++;
+        if (modeIdx >= MODES.length) {
+            modeIdx = 0;
+            shapeIdx++;
+            if (shapeIdx >= SHAPES.length) {
+                shapeIdx = 0;
+            }
+        }
+        SHAPE = SHAPES[shapeIdx];
+        MODE = MODES[modeIdx];
+        num = (shapeIdx * MODES.length + modeIdx) + 1;
+        redraw();
+    }
+
+    function keyPressed() {
+        nextSetting();
+    }
+
+    function mousePressed() {
+        nextSetting();
+    }
+
+    function ellipseCorners(x1, y1, x2, y2) {
+      // Bounding box (light gray)
+      rectMode(CORNERS);
+      stroke(200);
+      noFill();
+      strokeWeight(1);
+      rect(x1, y1, x2, y2);
+
+      // Adjust coordinates for testing modes other than CORNERS
+      if (MODE == CORNER) {
+        let x = min(x1, x2);  // x
+        let y = min(y1, y2);  // y
+        // Don't use abs on w and h, so we get negative values as well
+        let w = x2 - x1; // w
+        let h = y2 - y1; // h
+        x1 = x; y1 = y; x2 = w; y2 = h;
+      } else if (MODE == CENTER) {
+        let x = (x2 + x1) / 2; // x
+        let y = (y2 + y1) / 2; // y
+        // Don't use abs on w and h, so we get negative values as well
+        let w = x2 - x1;
+        let h = y2 - y1;
+        x1 = x; y1 = y; x2 = w; y2 = h;
+      } else if (MODE == RADIUS) {
+        let x = (x2 + x1) / 2; // x
+        let y = (y2 + y1) / 2; // y
+        // Don't use abs on w and h, so we get negative values as well
+        let r1 = (x2 - x1) / 2; // r1;
+        let r2 = (y2 - y1) / 2; // r2
+        x1 = x; y1 = y; x2 = r1; y2 = r2;
+      }
+
+
+      // P1, P2 dots
+      stroke(150);
+      strokeWeight(4);
+      point(x1, y1);
+      if (MODE === CORNERS) {
+        point(x2, y2);
+      }
+
+      // P1, P2 text labels
+      fill(150);
+      noStroke();
+      text('P1', x1 + 5, y1);
+      if (MODE === CORNERS) {
+        text('P2', x2 + 5, y2);
+      }
+
+
+      // Shape
+      noFill();
+      stroke(0);
+      strokeWeight(1);
+      ellipseMode(MODE);
+      rectMode(MODE);
+
+      try {
+        console.log(`Drawing (${x1}, ${y1}, ${x2}, ${y2})`);
+        if (SHAPE === 'ellipse') {
+            ellipse(x1, y1, x2, y2);
+        } else if (SHAPE === 'arc') {
+            const GAP = 0.1;
+            arc(x1, y1, x2, y2, 0 + GAP/2, HALF_PI - GAP);
+            arc(x1, y1, x2, y2, HALF_PI + GAP, PI - GAP);
+            arc(x1, y1, x2, y2, PI + GAP, PI + HALF_PI - GAP);
+            arc(x1, y1, x2, y2, PI + HALF_PI + GAP, TWO_PI - GAP);
+        } else if (SHAPE === 'rect') {
+            rect(x1, y1, x2, y2);
+        }
+      } catch (e) {
+        console.warn(`Error drawing (${x1}, ${y1}, ${x2}, ${y2})`);
```

Comment:
If you are trying to write tests for it? You could probably refer to this file, this could help you to write tests for it? 

https://github.com/processing/p5.js/blob/main/contributor_docs/unit_testing.md

Let me know if this file is not related to testing purpose.